openshift/generate-manifests.sh

```diff
@@ -61,6 +61,8 @@ mkdir -p "$TMP_MANIFEST_DIR"
 # Run kustomize, which emits a single yaml file
 TMP_KUSTOMIZE_OUTPUT="${TMP_MANIFEST_DIR}/temp.yaml"
 $KUSTOMIZE build "${TMP_ROOT}/openshift/kustomize/overlays/openshift" -o "$TMP_KUSTOMIZE_OUTPUT"
+echo "---" >> "$TMP_KUSTOMIZE_OUTPUT"
+$KUSTOMIZE build "${TMP_ROOT}/openshift/kustomize/overlays/openshift/resources/catalogs" >> "$TMP_KUSTOMIZE_OUTPUT"
```

Comment:
Why not include this in the openshift overlay?

Reply:
I originally tried this. I hit a snag with kustomize behaviors overriding the namespace and name that I wanted these resources to have and I was (and still am) prioritizing working on https://github.com/openshift/cluster-olm-operator/pull/73 . To test the work I'm doing over in cluster-olm-operator I needed _something_ resembling what we wanted and this seemed like a good method to get there quickly. 

I intend to come back to this and investigate the snag a bit further after I'm satisfied with the state of my cluster-olm-operator PR.

For more details on the snag, including these in our general overlay resulted in:
- using `catalogd-` as the prefix in the names instead of `openshift-`. We could potentially fix this by making the overlay prefix `openshift-catalogd-` (or even just `openshift-`) but I haven't spent more time looking into how this would impact the existing resource names (and maybe it doesn't matter).
- setting the namespace on the `ClusterCatalog` resources because kustomize isn't aware it is a cluster-scoped resource. Not a huge deal since kubernetes will ignore the namespace for cluster-scoped resources, but it feels clunky.

Reply:
Got it. +1 on avoiding both of those issues, but there's a simple way to do that. @m1kola added new kustomizations in the PR that gives catalogd permission to list/watch secrets in the openshift-config namespace. I think that would serve as a good reference for how to separate out things to avoid sharing namespace and namePrefix settings.